packages/component/src/SendBox/SuggestedActions.js

```diff
@@ -69,28 +62,65 @@ const SuggestedActions = ({ className, suggestedActions = [] }) => {
     />
   ));
 
-  if (suggestedActionLayout === 'stacked') {
-    return (
-      <div aria-label=" " aria-live="polite" role="status">
+  const renderSuggestedActionsEmpty = useCallback(() => <ScreenReaderText text={suggestedActionsContainerText} />, [
+    suggestedActionsContainerText
+  ]);
+
+  const renderSuggestedActionsStacked = useCallback(
+    () => (
+      <React.Fragment>
         <ScreenReaderText text={suggestedActionsContainerText} />
         <div className={classNames(suggestedActionsStyleSet + '', SUGGESTED_ACTION_STACKED_CSS + '', className + '')}>
           {children}
         </div>
-      </div>
-    );
-  }
+      </React.Fragment>
+    ),
+    [children, className, suggestedActions, suggestedActionsContainerText, suggestedActionsStyleSet]
+  );
+
+  const renderSuggestedActionsContent = useCallback(
+    () => (
+      <React.Fragment>
+        <ScreenReaderText text={suggestedActionsContainerText} />
+        <BasicFilm
+          autoCenter={false}
+          className={classNames(suggestedActionsStyleSet + '', className + '')}
+          showDots={false}
+          styleSet={suggestedActionsStyleSetForReactFilm}
+        >
+          {children}
+        </BasicFilm>
+      </React.Fragment>
+    ),
+    [
+      children,
+      className,
+      suggestedActions,
+      suggestedActionsContainerText,
+      suggestedActionsStyleSet,
+      suggestedActionsStyleSetForReactFilm
+    ]
+  );
+
+  const renderSuggestedActionsContainer = useCallback(() => {
+    if (!suggestedActions.length) {
+      return renderSuggestedActionsEmpty();
+    } else if (suggestedActionLayout === 'stacked') {
+      return renderSuggestedActionsStacked();
+    }
+    return renderSuggestedActionsContent();
+  }, [
+    renderSuggestedActionsContent,
+    renderSuggestedActionsEmpty,
+    renderSuggestedActionsStacked,
+    suggestedActionLayout,
+    suggestedActions
+  ]);
 
   return (
-    <div aria-label=" " aria-live="polite" role="status">
-      <ScreenReaderText text={suggestedActionsContainerText} />
-      <BasicFilm
-        autoCenter={false}
-        className={classNames(suggestedActionsStyleSet + '', className + '')}
-        showDots={false}
-        styleSet={suggestedActionsStyleSetForReactFilm}
-      >
-        {children}
-      </BasicFilm>
+    // aria-live is implicitly set to polite when role="status"
+    <div aria-label=" " role="status">
+      {renderSuggestedActionsContainer()}
```

Comment:
I think something like this would be simpler.

```javascript
  return (
    <div aria-label=" " aria-live="polite" role="status">
      <ScreenReaderText text={suggestedActionsContainerText} />
      {suggestedActions.length && (suggestedActionLayout === 'stacked' ? (
        <div className={classNames(suggestedActionsStyleSet + '', SUGGESTED_ACTION_STACKED_CSS + '', className + '')}>
          {children}
        </div>
      ): (
        <BasicFilm
          autoCenter={false}
          className={classNames(suggestedActionsStyleSet + '', className + '')}
          showDots={false}
          styleSet={suggestedActionsStyleSetForReactFilm}
        >
          {children}
        </BasicFilm>
      )
      )}
    </div>
  );
```

Reply:
Same to me.

Try very best not to write a render function. Decompositing component could help.